static/js/unread.js

```diff
@@ -260,7 +259,7 @@ exports.num_unread_for_person = function (user_ids_string) {
     return unread_privates.get(user_ids_string).num_items();
 };
 
-exports.update_unread_counts = function () {
+exports.update_unread_counts = function (stream_list_update_dom_with_unread_counts) {
```

Comment:
I feel like we can break a lot of dependencies here by moving update_unread_counts() to unread_ui.js, so we don't have to pass in all these functions.